components/layout/fragment.rs

```diff
@@ -1079,6 +1080,25 @@ impl Fragment {
             }
         };
 
+        // Inline fragments that are part of a text run don't have interior borders.
+        // So don't include interior borders when calculating positioning.
+        match self.specific {
+            SpecificFragmentInfo::ScannedText(ref txt_info) => {
+                let (begin, end, run_len) = match (txt_info.range.begin(),
+                                                   txt_info.range.end(),
+                                                   txt_info.run.char_len()) {
+                    (CharIndex(begin), CharIndex(end), CharIndex(run_len)) => (begin, end, run_len)
+                };
+                if begin > 0 {
+                    border.inline_start = Zero::zero();
+                }
+                if end < run_len {
+                    border.inline_end = Zero::zero();
+                }
+            },
+            _ => ()
+        };
```

Comment:
This should be factored out into a separate function and shared between DL building and here. (Part of `Fragment` is fine.)